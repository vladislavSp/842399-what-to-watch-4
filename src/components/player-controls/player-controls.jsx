import React from "react";
import PropTypes from "prop-types";
import {withStateProgress} from "../../hocs/withStateProgress.jsx";

export const PlayerControls = (props) => {
  const {remainingTime, progress} = props;

  return (
    <div className="player__controls-row">
      <div className="player__time">
        <progress
          className="player__progress"
          value={progress}
          max="100"
        ></progress>
        <div className="player__toggler" style={{left: `${progress}%`}}>
          Toggler
        </div>
      </div>
      <div className="player__time-value">{remainingTime}</div>
    </div>
  );
};

PlayerControls.propTypes = {
  progress: PropTypes.number.isRequired,
  remainingTime: PropTypes.string.isRequired
};

export default withStateProgress(PlayerControls);
