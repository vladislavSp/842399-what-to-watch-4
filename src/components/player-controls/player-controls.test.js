import React from "react";
import renderer from "react-test-renderer";
import {PlayerControls} from "./player-controls.jsx";

it(`Player controls renders correctly`, () => {
  const tree = renderer
    .create(<PlayerControls remainingTime={`00:50`} progress={10} />, {
      createNodeMock: () => {
        return {};
      }
    })
    .toJSON();
  expect(tree).toMatchSnapshot();
});
