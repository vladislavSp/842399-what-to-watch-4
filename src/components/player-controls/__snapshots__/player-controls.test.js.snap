// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player controls renders correctly 1`] = `
<div
  className="player__controls-row"
>
  <div
    className="player__time"
  >
    <progress
      className="player__progress"
      max="100"
      value={10}
    />
    <div
      className="player__toggler"
      style={
        Object {
          "left": "10%",
        }
      }
    >
      Toggler
    </div>
  </div>
  <div
    className="player__time-value"
  >
    00:50
  </div>
</div>
`;
